src/writer/queue.h

```diff
@@ -80,6 +81,18 @@ void Queue<T>::pushToQueue(const T &element) {
     pthread_mutex_unlock(&m_queueMutex);
 }
 
+template<typename T>
+bool Queue<T>::getHead(T &element) {
+  pthread_mutex_lock(&m_queueMutex);
+  if (m_realQueue.empty()) {
+    pthread_mutex_unlock(&m_queueMutex);
```

Comment:
Probably because we start using thread before using C++11 :)

However, I remember that some C++ implementation miss some functionality about thread. I don't remember witch one exactly. But I would point mingw implementation and android.

It would be nice to use c++11 stl's multithreading library instead of pthread. It would be os independent (and we could avoid using a pthread wrapping library on windows). But it is probably for another PR :)